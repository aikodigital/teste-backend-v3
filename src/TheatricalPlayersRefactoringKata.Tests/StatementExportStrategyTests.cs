using ApprovalTests;
using ApprovalTests.Reporters;
using TheatricalPlayersRefactoringKata.Domain.Enums;
using TheatricalPlayersRefactoringKata.Domain.Models;
using TheatricalPlayersRefactoringKata.Infrastructure.Strategies.Exports;

namespace TheatricalPlayersRefactoringKata.Tests;


public class StatementExportStrategyTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TxtStatementExportStrategy_ShouldExportCorrectly()
    {
        // Arrange
        var customer = new Customer("John Doe");

        var performances = new List<Performance>
        {
            new Performance(new Play("Comedy Play", GenreEnum.Comedy, 2000), 25) { Amount = 300, Credits = 30 },
            new Performance(new Play("Tragedy Play", GenreEnum.Tragedy, 2000), 35) { Amount = 400, Credits = 40 }
        };

        var invoice = new Invoice(customer, performances);
        var statement = new TxtStatementExportStrategy();

        // Act
        var result = statement.GenerateStatement(invoice);

        // Assert
        Approvals.Verify(result);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void XmlStatementExportStrategy_ShouldExportCorrectly()
    {
        // Arrange
        var customer = new Customer("John Doe");

        var performances = new List<Performance>
        {
            new Performance(new Play("Comedy Play", GenreEnum.Comedy, 2000), 25) { Amount = 300, Credits = 30 },
            new Performance(new Play("Tragedy Play", GenreEnum.Tragedy, 2000), 35) { Amount = 400, Credits = 40 }
        };

        var invoice = new Invoice(customer, performances);
        var statement = new XmlStatementExportStrategy();

        // Act
        var result = statement.GenerateStatement(invoice);

        // Assert
        Approvals.Verify(result);
    }
}