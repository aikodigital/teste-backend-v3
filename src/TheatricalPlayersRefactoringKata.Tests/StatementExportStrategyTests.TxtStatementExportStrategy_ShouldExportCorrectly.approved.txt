Statement for John Doe
 Comedy Play: $300.00 (25 seats)
 Tragedy Play: $400.00 (35 seats)
Amount owed is $700.00
You earned 70 credits
